File: MotorImageryProcessing/src/main/java/ai/hellomoto/mip/common_fragments/CustomLineChart.java
package ai.hellomoto.mip.common_fragments;

import javafx.beans.NamedArg;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.scene.chart.Axis;
import javafx.scene.chart.LineChart;

public class CustomLineChart<X, Y> extends LineChart<X, Y> {
    public CustomLineChart(@NamedArg("xAxis") Axis<X> xAxis, @NamedArg("yAxis") Axis<Y> yAxis) {
        this(xAxis, yAxis, FXCollections.<Series<X, Y>>observableArrayList());
    }

    public CustomLineChart(@NamedArg("xAxis") Axis<X> xAxis, @NamedArg("yAxis") Axis<Y> yAxis, @NamedArg("data") ObservableList<Series<X,Y>> data) {
        super(xAxis,yAxis);
        setData(data);
    }

    public void hideXAxis() {
        getChartChildren().remove(this.getXAxis());
    }
}

File: MotorImageryProcessing/src/main/kotlin/ai/hellomoto/mip/common_fragments/CustomLineChart.kt
package ai.hellomoto.mip.common_fragments

import javafx.geometry.Side
import javafx.scene.chart.CategoryAxis
import javafx.scene.chart.LineChart
import javafx.scene.chart.NumberAxis


class CustomLineChart : LineChart<String, Number>(
    CategoryAxis().apply {
        animated = false
        side = Side.BOTTOM
        prefHeight = 0.0
        isTickMarkVisible = false
    },
    NumberAxis().apply {
        animated = false
        side = Side.LEFT
        isForceZeroInRange = false
    }
) {
    init {
        createSymbols = false
        animated = false
        verticalGridLinesVisible = false
        // style = "-fx-border-color: red"
    }

    fun hideXAxis(): CustomLineChart {
        chartChildren.remove(this.xAxis)
        return this
    }
}